packages/back-end/test/integrations/informationSchemas.test.ts

```diff
@@ -0,0 +1,453 @@
+import { RawInformationSchema } from "../../src/types/Integration";
+import { formatInformationSchema } from "../../src/util/integrations";
+
+describe("formatInformationSchema", () => {
+  it("Correctly formats a rawInformationSchema for BigQuery correctly", () => {
+    const rawInformationSchema: RawInformationSchema[] = [
+      {
+        table_catalog: "adept-arbor-354914",
+        table_name: "experiment-assignments",
+        table_schema: "a_second_data_set",
+        column_count: "3",
+      },
+      {
+        table_catalog: "adept-arbor-354914",
+        table_name: "experiment-assignments",
+        table_schema: "a_second_data_set",
+        column_count: "3",
+      },
+      {
+        table_catalog: "adept-arbor-354914",
+        table_name: "experiment-assignments",
+        table_schema: "a_second_data_set",
+        column_count: "3",
+      },
+      {
+        table_catalog: "adept-arbor-354914",
+        table_name: "orders",
+        table_schema: "sample_data_set",
+        column_count: "3",
+      },
+      {
+        table_catalog: "adept-arbor-354914",
+        table_name: "orders",
+        table_schema: "sample_data_set",
+        column_count: "3",
+      },
+      {
+        table_catalog: "adept-arbor-354914",
+        table_name: "orders",
+        table_schema: "sample_data_set",
+        column_count: "3",
+      },
+      {
+        table_catalog: "adept-arbor-354914",
+        table_name: "page-visitors",
+        table_schema: "sample_data_set",
+        column_count: "3",
+      },
+      {
+        table_catalog: "adept-arbor-354914",
+        table_name: "page-visitors",
+        table_schema: "sample_data_set",
+        column_count: "3",
+      },
+      {
+        table_catalog: "adept-arbor-354914",
+        table_name: "page-visitors",
+        table_schema: "sample_data_set",
+        column_count: "3",
+      },
+      {
+        table_catalog: "adept-arbor-354914",
+        table_name: "sample_table",
+        table_schema: "sample_data",
+        column_count: "3",
+      },
+      {
+        table_catalog: "adept-arbor-354914",
+        table_name: "sample_table",
+        table_schema: "sample_data",
+        column_count: "3",
+      },
+      {
+        table_catalog: "adept-arbor-354914",
+        table_name: "sample_table",
+        table_schema: "sample_data",
+        column_count: "3",
+      },
+    ];
+    const date = new Date();
```

Comment:
Does anyone have suggestions on a better way to handle the date here? The `formatInformationSchema` sets the `dateCreated` and `dateUpdated` fields on a lot of the nested elements - this will later allow us to use a stale-while-revalidating caching method, but it introduces some challenges here.

Reply:
I tend to look only at properties I care about rather than check equality for the whole object, but that makes the assertions more verbose. For example:

```ts
expect(formattedResults.schemas[0].path).toEqual("`adept-arbor-354914.a_second_data_set`")
```

Same for other relevant properties, e.g. `schemas[].tables[].path`, `schemas[].schemaName`, etc. And because you can assume all schemas would be treated the same, testing every index is not necessary.

If you'd rather not do it that way and you want to do something like control time, you can look into [Jest timer mocks](https://jestjs.io/docs/timer-mocks) but that doesn't sound worth it for created and updated timestamps.

Reply:
Oooh - that's a great pattern. Thank you!